app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/db/entity/VideoEntity.java

```diff
@@ -0,0 +1,475 @@
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+@Entity(tableName = DatabaseContract.VideoEntry.TABLE_NAME)
+public class VideoEntity implements  Parcelable {
+
+    @SerializedName(GsonContract.DESCRIPTION)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_DESC)
+    private String mDescription;
+
+    @SerializedName(GsonContract.VIDEO_URLS)
+    @Ignore
+    private List<String> mVideoUrls;
+
+    @SerializedName(GsonContract.CARD_IMAGE_URL)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_CARD_IMAGE_URL)
+    private String mCardImageUrl;
+
+    @SerializedName(GsonContract.BACKGROUND_IMAGE_URL)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_BG_IMAGE_URL)
+    private String mBgImageUrl;
+
+    @SerializedName(GsonContract.TITLE)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_NAME)
+    private String mTitle;
+
+    @SerializedName(GsonContract.STUDIO)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_STUDIO)
+    private String mStudio;
+
+    /**
+     * The following field cannot be instantiate from json file.
+     */
+    // Primary key to store the video item.
+    @PrimaryKey(autoGenerate = true)
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_AUTO_GENERATE_ID)
+    private long mId;
+
+    // The url points to the local storage of the video content.
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_VIDEO_CACHE)
+    private String mVideoLocalStorageUrl;
+
+    // The url points to the local storage of the video background image.
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_BG_IMAGE_CACHE)
+    private String mVideoBgImageLocalStorageUrl;
+
+    // The url points to the local storage of the video card image.
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_CARD_IMG_CACHE)
+    private String mVideoCardImageLocalStorageUrl;
+
+    // The mCategory of current video
+    // A single mCategory may contain many different videos
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_CATEGORY)
+    private String mCategory;
+
+    // The first url in urls array
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_VIDEO_URL)
+    private String mVideoUrl;
+
+    public String getTrailerVideoUrl() {
+        return mTrailerVideoUrl;
+    }
+
+    public void setTrailerVideoUrl(String mTrailerVideoUrl) {
+        this.mTrailerVideoUrl = mTrailerVideoUrl;
+    }
+
+    // The first url in urls array
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_VIDEO_TRAILER_URL)
+    private String mTrailerVideoUrl;
+
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_VIDEO_IS_RENTED)
+    private boolean mRented;
+
+    @ColumnInfo(name = DatabaseContract.VideoEntry.COLUMN_VIDEO_STATUS)
+    private String mStatus;
+
+    /**
+     * The constructor is requried by room database
+     */
+    public VideoEntity() {
```

Comment:
ditto.

Reply:
Hi Yigit, please correct me if I'm wrong. But VideoEntity has implemented the Parcelable interface which means there is already a constructor declared as` protected VideoEntity(Parcel in).` So I guess it is necessary to keep this empty constructor for room to use. And from my personal experiment, if I remove this constructor, the program won't get compiled.

Reply:
yes, i missed the parcel constructor, sorry.